blog/content/post/2020-07-21-secure-app-coding-practices.md

```diff
@@ -0,0 +1,74 @@
+Remember - always verify the data also the second time, on the server side.
+
+### Try… catch mistakes
+
+Working with external libraries, these may have different error notification systems. Some just "throw" Error, others print exact messages, and some let us decide ourselves how to behave in case of a problem. However, we do not want JS throwing the default error content to the user, because it is often incomprehensible and can expose us to leakage of sensitive data. Remember to use the Try… catch block to contol the errors.
+
+### Keep your web app updated
+
+Try to use popular and valued packages. Pay attention to the number of editions, downloads and the date of the last update. It is very important that the package is still supported. Then, even if any gaps are found, they should be patched quickly.
+
+Avoid packages that the author himself describes as deprecated. Even if you think a package is perfect for your needs, it's quite possible that in many cases it won't work properly with your code. In addition, there is a good chance that it has some security flaws. Remember that IT is an extremely dynamic industry.
+Update your dependent packages
+Also to avoid vulnerabilities that are present in any framework or library, you should make sure you actually use all the libraries integrated into your software and use the latest version of each library, if it’s stable.
+
+### Tools that may be useful to you
+
+If any gaps are found, the authors of the packages try to patch them. In order for your app to take advantage of these changes (fixes), you need to update them as needed. Unfortunately, we often have so many of them that checking each of them manually would be quite a breakneck task. So a good solution may be to use a tool called **Snyk**. It automatically checks the GitHub repository and reports on possible dangers. You can use it online at [this link](https://snyk.io/test/).
+
+In addition to the above good practices, the use of the **helmet** package is standard. Its task is to properly set the HTTP headers so that our server is less vulnerable to attacks. A description of all functionalities can be found at [here](https://helmetjs.github.io/). Take this package as a basis and something that will as much as possible hamper attacks that could take advantage of the weaknesses of the header settings.
+
+
+## Summary
+
+The more extensive and complex the application, the more vulnerabilities may appear init. If its role is sensitive (e.g. creating a bank's website), we will be exposed to more attacks, and these can often be extremely sophisticated. So we are not able to prepare ourselves for a threat in 100%.  Nevertheless, we must try to minimize the danger. Securing your applications is very important and the industry is constantly evolving. Important systems must be analyzed on an on going basis and, if necessary, adjusted to deal with new forms of attacks. Only this approach gives us are latively large guarantee as to safety.
```

Comment:
We can mention validation steps or TDD culture as a means to achieve better security and integration within our app

Reply:
I would like to address the issue of injections, neutralizing threats, "broken authentication" and testing  in another article because the topic is really extensive. Here I only focused on the basic practices of secure coding

Reply:
the application is in it/inside.